app/src/main/java/org/mozilla/reference/browser/browser/BrowserAutocompleteProvider.kt

```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// This is, roughly, an "auto-complete aggregator" - it first tries to find a match via "places",
+// and if that fails, falls back to the "domain" provider.
+package org.mozilla.reference.browser.browser
+
+import android.app.Activity
+import android.content.Context
+import kotlinx.coroutines.experimental.launch
+import mozilla.components.browser.domains.DomainAutoCompleteProvider
+import mozilla.components.browser.toolbar.BrowserToolbar
+import mozilla.components.support.base.log.Log
+import mozilla.components.ui.autocomplete.InlineAutocompleteEditText
+import java.net.URL
+import org.mozilla.places.PlacesConnection
+
+
+fun debug(message: String) {
+    Log.log(tag="BrowserAutocomplete", message=message);
+}
+
+class BrowserAutocompleteProvider (
```

Comment:
I think this type should ultimately live in the components repo. Likely, part of this component:
https://github.com/mozilla-mobile/android-components/issues/1109

Reply:
`PlacesConnection` could live in the services/sync-places component using the `mozilla.components` package.

Reply:
PlacesConnection has an unsafe implementation so it should live in a-s. (Although maybe there will be a very similar type in mozilla.components, as with the recent fxa work)